### PR TITLE
[ADF-4496] Share dialog enhancements

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- raise an error snackbar when an error happens during unshare (i.e. permission error)
- allow "unsetting" the expiration date
- possibility to choose between showing only date or date and time
- extra tests

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4496

### CODE REVIEW

File: lib/content-services/content-node-share/content-node-share.dialog.ts [hunk 1]
```diff
@@ -68,21 +80,26 @@ export class ShareDialogComponent implements OnInit, OnDestroy {
         private nodesApiService: NodesApiService,
         private contentService: ContentService,
         private renditionService: RenditionsService,
-        @Inject(MAT_DIALOG_DATA) public data: any) {
-    }
+        private notificationService: NotificationService,
```
Comment (eromano) [inline lib/content-services/content-node-share/content-node-share.dialog.ts]: didn't we decide to not use anymore the NotificationService is the component but wires it in the apps?

File: lib/content-services/content-node-share/content-node-share.dialog.ts [hunk 2]
```diff
@@ -123,7 +138,27 @@ export class ShareDialogComponent implements OnInit, OnDestroy {
...
 
     get canUpdate() {
-        return this.contentService.hasAllowableOperations(this.data.node.entry, 'update');
+        return this.contentService.hasAllowableOperations(
+            this.data.node.entry,
+            'update'
+        );
+    }
+
+    onToggleExpirationDate(slideToggle) {
```
Comment (eromano) [inline lib/content-services/content-node-share/content-node-share.dialog.ts]: missing type

File: lib/content-services/content-node-share/content-node-share.dialog.ts (hunk 1 above)
Comment (DenysVuika) [reply lib/content-services/content-node-share/content-node-share.dialog.ts]: Ok, will move it to demo shell.

**Comment by DenysVuika** [comment]
@eromano updated code, the e2e seem to be green except cloud components (not related to this PR)

**Comment by DenysVuika** [comment]
@eromano the test was removed because corresponding functionality is removed, the PR removes the Time Picker from the Share

**Comment by eromano** [comment]
@DenysVuika there is a reason why do we have here do not merge?

**Comment by DenysVuika** [comment]
This PR removes the time picker. The corresponding Jira has more details and discussions

**Comment by eromano** [comment]
@DenysVuika I agree on making it optional we can have an input parameter that disables the date parameters

**Comment by DenysVuika** [comment]
Yeah, that’s why it’s marked as do not merge. Needs some extra work @pionnegru

**Comment by pionnegru** [comment]
@DenysVuika @eromano his pr has been updated.

**Comment by pionnegru** [comment]
I missed the input part. app config slipped in. I'll do the changes.

**Pull request merged**